openwisp_firmware_upgrader/upgraders/openwrt.py

```diff
@@ -155,8 +155,38 @@ def upload(self, image_file, remote_path):
         except Exception as e:
             raise RecoverableFailure(str(e))
 
+    def _verify_device_uuid(self):
+        """
+        Verifies that the UUID of the device being upgraded matches
+        the UUID in the device's configuration
+        """
+        device_uuid = self.upgrade_operation.device.pk
+        # Get UUID from device's openwisp config
+        output, exit_code = self.exec_command(
+            'uci get openwisp.http.uuid', exit_codes=[0, 1]
+        )
+        if exit_code == 1 or not output.strip():
+            self.log(_('Could not read device UUID from configuration'))
+            raise UpgradeAborted(
+                'Could not verify device identity: UUID not found in device configuration'
```

Comment:
I looked around the code base and I've noticed we always just call `raise UpgradeAborted()`, without any message, so I believe there's no need to specify an exception message here because we are already providing an error message by calling the `self.log()` method (it's redundant but with no benefits).